packages/xrpl/src/client/RequestManager.ts

```diff
@@ -206,7 +209,7 @@ export default class RequestManager {
     // status no longer needed because error is thrown if status is not "success"
     delete response.status
     // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- Must be a valid Response here
-    this.resolve(response.id, response as unknown as Response)
+    this.resolve(response.id, response as unknown as Response<APIVersion>)
```

Comment:
You shouldn't need to use type assertions here if you change `resolve` to be `Partial<Response<`